src/winrpath.cxx

```diff
@@ -496,6 +501,58 @@ void CoffParser::ParseData(PIMAGE_ARCHIVE_MEMBER_HEADER header, coff_member *mem
 }
 
 
+void CoffParser::NormalizeLinkerMember(
+    const std::string &name,
+    const int &offset,
+    const int &base_offset,
+    const char * strings,
+    const DWORD symbols
+)
+{
+    int offset_with_header = base_offset + sizeof(IMAGE_ARCHIVE_MEMBER_HEADER);
+    int current_relative_offset = 0;
+    for(int j=0; j<symbols;++j) {
+        int name_len = strlen(strings+current_relative_offset);
+        char * new_name = new char[name_len+1];
+        strcpy(new_name, strings+current_relative_offset);
+        if(strstr(new_name, name.c_str())) {
+            replace_special_characters(new_name, name_len);
+            int foffset = offset_with_header + offset + current_relative_offset;
+            this->coffStream->seek(0);
+            this->coffStream->seek(foffset);
+            this->coffStream->write(new_name, name_len);                        
+        }
+        current_relative_offset += name_len+1;
+        delete new_name;
+    }
+}
+
+void CoffParser::NormalizeSectionNames(const std::string &name, char* section, const DWORD &section_data_start_offset, int data_size)
+{
+    int name_len = name.size();
+    char * section_search_start = section;
+    char * search_terminator = section+data_size;
+    ptrdiff_t offset = 0;
+    while(section_search_start && (section_search_start < search_terminator)) {
+        section_search_start = findstr(section_search_start, name.c_str(), data_size-offset);
+        if (section_search_start) {
+            // we found a name, rename
+            offset = section_search_start - section;
+            char * new_name = new char[name_len+1];
+            strncpy(new_name, section_search_start, name_len+1);
```

Comment:
Should this be copying `name_len` characters vs. `name_len+1`.

Making the string `name_len+1` characters seems right, so that change (as part of refactoring) seems reasonable. 

Also, this seems like a separate issue from the core one mentioned, and for the most part this is refactoring some code into a function, did I get that right or is this a part of the core issue?

Reply:
Actually this probably isn't necessary at all, I made the change to debug (otherwise my output was full of whatever garbage memory happened to be at the end of that char array). I can pull it out/into my other branches.